File: Anul II/Medii de proiectare si programare/Laborator/lab06/services/src/main/java/com/ubb/service/UserService.java
package com.ubb.service;

import com.ubb.model.User;
import com.ubb.repository.IUserRepository;
import com.ubb.validation.IValidator;
import com.ubb.validation.UserValidator;

import java.util.Collection;
import java.util.Optional;

public class UserService implements IUserService {
    private final IUserRepository userRepository;
    private final IValidator<User> validator;

    public UserService(final IUserRepository userRepository) {
        this.userRepository = userRepository;
        validator = new UserValidator();
    }
    @Override
    public Optional<User> findById(Long id) {
        return userRepository.findById(id);
    }

    @Override
    public Collection<User> findAll() {
        return userRepository.findAll();
    }

    @Override
    public void save(User newEntity) {
        validator.validateEntity(newEntity);
        userRepository.save(newEntity);
    }

    @Override
    public void delete(Long id) {
        userRepository.delete(id);
    }

    @Override
    public void update(User updatedEntity) {
        validator.validateEntity(updatedEntity);
        userRepository.update(updatedEntity);
    }

    @Override
    public Optional<User> getLogin(String username, String password) {
        return userRepository.getByUsernameAndPassword(username, password);
    }
}

File: Anul II/Medii de proiectare si programare/Laborator/lab08/C#/Ubb.BikeContest.Service/src/Service/UserService.cs
using System.Collections;
using Ubb.BikeContest.Model;
using Ubb.BikeContest.Repository;
using Ubb.BikeContest.Services.validation;

namespace Ubb.BikeContest.Services.src.Service;

public class UserService : IUserService
{
    private readonly IUserRepository _userRepository;
    private readonly IValidator<long, User> _validator;

    public UserService(IUserRepository userRepository)
    {
        _userRepository = userRepository;
        _validator = new UserValidator();
    }

    public User FindById(long id)
    {
        return _userRepository.FindById(id);
    }

    public IEnumerable<User> FindAll()
    {
        return _userRepository.FindAll();
    }

    public void Save(User newEntity)
    {
        _validator.ValidateEntity(newEntity);
        _userRepository.Save(newEntity);
    }

    public void Delete(long id)
    {
        _userRepository.Delete(id);
    }

    public void Update(User updatedEntity)
    {
        _userRepository.Update(updatedEntity);
    }

    public User GetLogin(string username, string password)
    {
        return _userRepository.GetByUsernameAndPassword(username, password);
    }
}